torchinfo/layer_info.py

```diff
@@ -126,6 +127,28 @@ def nested_list_size(inputs: Sequence[Any]) -> tuple[list[int], int]:
 
         return size, elem_bytes
 
+    def get_is_trainable(self) -> str:
+        """
+        Checks if the module is trainable.
+
+        Returns:
+            "True", if all the parameters are trainable (`requires_grad=True`)
+            "False" if none of the parameters are trainable
+            "Partial" if otherwise. Say, weights are trainable but not bias.
```

Comment:
Do submodules matter for this column? I notice we do not have recurse=True in module.parameters(), so I'm curious whether we should enable it or not to get the correct answer.

Reply:
Sorry for the super late reply.
 
I have used `named_parameters()` as it was the most suggested way in the PyTorch forums (by the maintainers).
> the named_parameters() method does not look for all objects that are contained in your model, just the nn.Modules and nn.Parameters, so as I stated above, if you store you parameters outsite of these, then they won’t be detected by named_parameters(). 

https://discuss.pytorch.org/t/model-named-parameters-will-lose-some-layer-modules/14588/4

I think submodules are quite needed to check if parts of layers are frozen. `named_parameters()` and also `parameters()` have recurse=True by default. And will check all modules and submodules.
```
def named_parameters(self, prefix: str = '', recurse: bool = True) -> Iterator[Tuple[str, Parameter]]:
    ...
    recurse (bool): if True, then yields parameters of this module
        and all submodules. Otherwise, yields only parameters that
        are direct members of this module.
    ...
```